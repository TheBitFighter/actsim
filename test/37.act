function foo (int<32> in) : int<32>
{

	chp {
		self := ~in
	}

}

defproc test () 
{

	int b;

	chp {
		b := 2;
		log (b);
		b := foo(b);
		log (b);

		b := 0xFFFFFFFE;
		log (b);
		b := foo(b);
		log (b)
	}

}

test t;
