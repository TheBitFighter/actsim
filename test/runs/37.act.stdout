[      10 t#:0] <> 2
[      20 t#:0] <> 4294967294
[      30 t#:0] <> 4294967294
[      40 t#:0] <> 2
